src/ui/components/Scanner/Scanner.tsx

```diff
@@ -332,8 +331,8 @@ const Scanner = forwardRef(
     const initScan = async () => {
       if (isPlatform("ios") || isPlatform("android")) {
         const allowed = await checkPermission();
-        setPermisson(allowed);
-        onCheckPermissionFinish?.(allowed);
+        setPermisson(!!allowed);
```

Comment:
Not sure if I understand why you had to change this.

Reply:
Because result of `checkPermission` is `boolean | undefined` but parameter of `setPermisson` and `onCheckPermissionFinish` is `boolean` only. Therefore, I need convert result of `checkPermission` to avoid type error.